sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusMessageProcessor.java

```diff
@@ -34,24 +42,26 @@
     private final ClientLogger logger = new ClientLogger(ServiceBusMessageProcessor.class);
     private final boolean isAutoComplete;
     private final AmqpRetryOptions retryOptions;
-    private final Function<ServiceBusReceivedMessage, Mono<Void>> completeFunction;
-    private final Function<ServiceBusReceivedMessage, Mono<Void>> onAbandon;
-    private final Function<ServiceBusReceivedMessage, Mono<Instant>> onRenewLock;
+    private final AmqpErrorContext errorContext;
+    private final Function<MessageLockToken, Mono<Void>> completeFunction;
+    private final Function<MessageLockToken, Mono<Void>> onAbandon;
+    private final Function<MessageLockToken, Mono<Instant>> onRenewLock;
     private final Deque<ServiceBusReceivedMessage> messageQueue = new ConcurrentLinkedDeque<>();
     private final boolean isAutoRenewLock;
     private final Duration maxAutoLockRenewal;
     private final MessageLockContainer messageLockContainer;
 
     ServiceBusMessageProcessor(boolean isAutoComplete, boolean isAutoRenewLock, Duration maxAutoLockRenewal,
```

Comment:
One option could be calculate `isAutoRenewLock` here using `maxAutoLockRenewal` value and  remove this argument `isAutoRenewLock` being passed in. But  it is not must.

Reply:
I thought about that. But decoupling it seemed like a better approach to separate responsibility.